express-notes-api/server.js

```diff
@@ -0,0 +1,116 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+const middle = express.json();
+
+app.use(middle);
+
+async function reader() {
+  try {
+    let data = await readFile('data.json', 'utf8');
+    data = JSON.parse(data);
+    return data;
+  } catch (error) {
+    console.log('Error: ', error);
+  }
```

Comment:
You probably don't want this try/catch here. It would be better to put the try/catch around every endpoint instead of in here. If this function were to throw an exception, it would log, but there would be no way to stop execution and/or let the caller know that something went wrong!

It's usually a bad practice to simply catch and log an exception. You usually want to either handle it or show a message to the user.